pyop2/codegen/c/inverse.c

```diff
@@ -10,14 +11,33 @@ static PetscScalar work_buffer[BUF_SIZE*BUF_SIZE];
 
 static void inverse(PetscScalar* __restrict__ Aout, const PetscScalar* __restrict__ A, PetscBLASInt N)
 {
+    PetscLogEvent  USER_EVENT_memcpy;
+    PetscLogEvent  USER_EVENT_getrf;
+    PetscLogEvent  USER_EVENT_getri;
+    PetscClassId   classid;
+
+    PetscClassIdRegister("PyOP2InverseCallable",&classid);
+    PetscLogEventRegister("PyOP2InverseCallable_memcpy",classid,&USER_EVENT_memcpy);
+    PetscLogEventRegister("PyOP2InverseCallable_getrf",classid,&USER_EVENT_getrf);
+    PetscLogEventRegister("PyOP2InverseCallable_getri",classid,&USER_EVENT_getri);
```

Comment:
What we should do is make these events in Python once rather than every time inverse is called and then just get the handle in C.